### PR TITLE
Trim down main class and call the correct followup implementation

### LINKED ISSUE #11: The .py file referenced from Addon.xml should be as minimal as possible
To get best speed the default python file should be bare minimal and then call the main file from that. Reason is the main referenced one isn't kept compiled while the others are kept compiled

### PR DESCRIPTION
**Pull request by razzeee**
This tries to close https://github.com/xbmc/generator-kodi-addon/issues/11

@MartijnKaijser

### CODE REVIEW

File: generators/app/templates/resources/lib/service.py [hunk 1]
```diff
@@ -5,18 +5,19 @@
...
     monitor = xbmc.Monitor()
 
     while not monitor.abortRequested():
         # Sleep/wait for abort for 10 seconds
         if monitor.waitForAbort(10):
             # Abort was requested while waiting. We should exit
             break
-        logger.debug("hello addon! %s" % time.time())
\ No newline at end of file
+        logger.debug("hello addon! %s" % time.time())
```
Comment (enen92) [inline generators/app/templates/resources/lib/service.py:23]: %s operator is being deprecated in favor of str.format in python3. Not sure if it is a good idea to promote the usage of: `hello addon {0}".format(time.time()). See: https://docs.python.org/3.1/whatsnew/2.6.html#pep-3101

File: generators/app/templates/resources/lib/plugin.py [hunk 2]
```diff
@@ -13,16 +14,21 @@
 kodiLogging.config()
 plugin = routing.Plugin()
 
+
 @plugin.route('/')
 def index():
-    addDirectoryItem(plugin.handle, plugin.url_for(show_category, "one"), ListItem("Category One"), True)
-    addDirectoryItem(plugin.handle, plugin.url_for(show_category, "two"), ListItem("Category Two"), True)
+    addDirectoryItem(plugin.handle, plugin.url_for(
+        show_category, "one"), ListItem("Category One"), True)
```
Comment (enen92) [inline generators/app/templates/resources/lib/plugin.py:21]: In my opinion (not that it matters much :) ), the fact that we instantiate and pass `Listitem` objects directly to the `addDirectoryItem` makes it really simple for new users to understand how to create a new directory item but hides a lot of logic that might be applied to listitems. Having the directory creation like above, will probably make users to use `iconImage` and `thumbnailImage` inline with the object instantiation. Both are deprecated and we making an effort for them to use `setArt` instead. Also, despite there are no actual benchmarks, some team members state that `addDirectoryItems` is much faster than `addDirectoryItem` when the amount of listitems to add is considerable. So probably a good idea to individually create them and just call `addDirectoryItems` with a list of `ListItems` (or a generator if supported).

File: generators/app/templates/resources/lib/service.py (hunk 1 above)
Comment (MartijnKaijser) [reply generators/app/templates/resources/lib/service.py:23]: If it works why not suggest/promote

**Comment by razzeee** [comment]
@enen92 would you be so kind and create issues or even prs for your suggestions?

**Comment by enen92** [comment]
Sure @Razzeee , I'll submit 2 different PR's once this is merged

**Comment by MartijnKaijser** [comment]
go for it. I'm sure we'll need several round anyway to get everything "perfect"

**Pull request merged**